react/provider/providerStore.tsx

```diff
@@ -72,6 +67,8 @@ const ProviderStore: FC = (props) => {
   }
 
   function validationValue() {
+    document.getElementById('div-border-color')?.classList.add('border-red')
```

Comment:
You is manipulating the DOM inside your provider, it's a bad practice to do.
My suggestion is create a state in your provider to indicate if `isValid` and get it in your component that you want remove or add `class`.